_assets/scripts/test_stats.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+import glob
+import xml.etree.ElementTree as ET
+test_stats = {}
+
+for file in glob.glob("**/report.xml"):
+    tree = ET.parse(file)
+    root = tree.getroot()
+    for testcase in root.iter("testcase"):
+        test_name = testcase.attrib["name"]
+
+        if test_name not in test_stats:
+            test_stats[test_name] = {"failed": 0, "total": 0}
+
+        test_stats[test_name]["total"] += 1
```

Comment:
You could use [setdefault](https://docs.python.org/2/library/stdtypes.html#dict.setdefault) for this.
```suggestion
        test_stats.setdefault(test_name, {"failed": 0, "total": 0})
        test_stats[test_name]["total"] += 1
```
But there is an even more Pythonic way using `defaultdict`:
```python
In [1]: from collections import defaultdict

In [2]: d = defaultdict(lambda: defaultdict(int))

In [3]: d['x']['z'] += 1

In [4]: d['x']['z'] += 1

In [5]: d['x']['y'] += 1

In [6]: d
Out[6]: 
defaultdict(<function __main__.<lambda>()>,
            {'x': defaultdict(int, {'z': 2, 'y': 1})})
```

Reply:
Beautiful, thanks!